apps/judicial-system/web/src/routes/Court/Indictments/Advocates/SelectDefender.tsx

```diff
@@ -14,106 +14,111 @@ import {
-        defenderNationalId: defendantWaivesRightToCounsel
-          ? null
-          : defendant.defenderNationalId || null,
-        defenderName: defendantWaivesRightToCounsel
-          ? null
-          : defendant.defenderName,
-        defenderEmail: defendantWaivesRightToCounsel
-          ? null
-          : defendant.defenderEmail,
-        defenderPhoneNumber: defendantWaivesRightToCounsel
-          ? null
-          : defendant.defenderPhoneNumber,
-        defenderChoice:
-          defendantWaivesRightToCounsel === true
-            ? DefenderChoice.WAIVE
-            : DefenderChoice.DELAY,
-      }
-
-      setAndSendDefendantToServer(updateDefendantInput, setWorkingCase)
-    },
-    [setWorkingCase, setAndSendDefendantToServer],
-  )
+  const handleUpdateDefendantState = (update: UpdateDefendant) => {
+    updateDefendantState({ caseId: workingCase.id, ...update }, setWorkingCase)
+  }
 
-  const toggleDefenderChoiceConfirmed = useCallback(
-    (
-      caseId: string,
-      defendant: Defendant,
-      isDefenderChoiceConfirmed: boolean,
-    ) => {
-      const { defenderChoice, defenderName } = defendant
-
-      const isDelaying =
-        !defenderName &&
-        (!defenderChoice || defenderChoice === DefenderChoice.CHOOSE)
-      const isChoosing =
-        defenderName &&
-        (!defenderChoice || defenderChoice === DefenderChoice.DELAY)
-
-      const defenderChoiceUpdate = isDelaying
-        ? DefenderChoice.DELAY
-        : isChoosing
-        ? DefenderChoice.CHOOSE
-        : defenderChoice
-
-      const updateDefendantInput = {
-        caseId,
-        defendantId: defendant.id,
-        isDefenderChoiceConfirmed,
-        defenderChoice: defenderChoiceUpdate,
-      }
-
-      setAndSendDefendantToServer(updateDefendantInput, setWorkingCase)
-      setDisplayModal(false)
-    },
-    [setWorkingCase, setAndSendDefendantToServer, setDisplayModal],
-  )
+  const handleUpdateDefendant = (update: UpdateDefendant) => {
+    updateDefendant({ caseId: workingCase.id, ...update })
+  }
 
-  const toggleCaseFilesSharedWithDefender = useCallback(
-    (
-      caseId: string,
-      defendant: Defendant,
-      caseFilesSharedWithDefender: boolean,
-    ) => {
-      const updateDefendantInput = {
-        caseId: caseId,
-        defendantId: defendant.id,
-        caseFilesSharedWithDefender,
-      }
-      setAndSendDefendantToServer(updateDefendantInput, setWorkingCase)
-    },
-    [setWorkingCase, setAndSendDefendantToServer],
-  )
+  const handleSetAndSendDefendantToServer = (update: UpdateDefendant) => {
+    setAndSendDefendantToServer(
+      { caseId: workingCase.id, ...update },
+      setWorkingCase,
+    )
+  }
+
+  const toggleDefendantWaivesRightToCounsel = (
+    defendant: Defendant,
+    defendantWaivesRightToCounsel: boolean,
+  ) => {
+    const update: UpdateDefendant = {
+      defendantId: defendant.id,
+      defenderNationalId: defendantWaivesRightToCounsel
```

Comment:
to avoid checking `defentantWaivesRightToCounsel` repeatedly, could we combine this in the following object? 
```
    ...(!defendantWaivesRightToCounsel
        ? {
            defenderNationalId: defendant.defenderNationalId,
            defenderName: defendant.defenderName,
            defenderEmail: defendant.defenderEmail,
            defenderPhoneNumber: defendant.defenderPhoneNumber,
            defenderChoice: DefenderChoice.WAIVE,
          }
        : { defenderChoice: DefenderChoice.DELAY }),
```
But this would mean that the null assignment of the properties would be `undefined`, but maybe that is sufficient?

Reply:
This code is not part of changes in this PR, it was only moved. We could do what you suggest, but we would need to set all the fields to null in the else clause. The value null has a different meaning than undefined: null makes it all the way to the database, clearing the database field, but undefined is ignored. I am not going to change this now, but we could do it in future PRs.

Reply:
Ah makes sense, thanks 🙏